### PR TITLE
Implement GetHosts, GetPorts, add tests and revamp

### PR DESCRIPTION
**Pull request by hkaj**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/collector/listeners/docker.go [hunk 1]
```diff
@@ -329,21 +307,57 @@ func (s *DockerService) GetADIdentifiers() ([]string, error) {
...
-	return s.Ports, nil
-}
+	if s.Ports != nil {
+		return s.Ports, nil
+	}
+
+	ports := make([]int, 0)
+	c := docker.Container{ID: string(s.ID)}
+
+	cInspect, err := c.Inspect(false)
```
Comment (mfpierre) [inline pkg/collector/listeners/docker.go]: Probably better to directly use `docker.Inspect(string(s.ID), false)` in this case

File: pkg/collector/listeners/docker.go [hunk 2]
```diff
@@ -329,21 +307,57 @@ func (s *DockerService) GetADIdentifiers() ([]string, error) {
...
 func (s *DockerService) GetHosts() (map[string]string, error) {
-	return s.Hosts, nil
+	if s.Hosts != nil {
+		return s.Hosts, nil
+	}
+
+	ips := make(map[string]string)
+	c := docker.Container{ID: string(s.ID)}
+
+	cInspect, err := c.Inspect(false)
```
Comment (mfpierre) [inline pkg/collector/listeners/docker.go]: Probably better to directly use docker.Inspect(string(s.ID), false) in this case

File: pkg/collector/listeners/docker.go [hunk 3]
```diff
@@ -329,21 +307,57 @@ func (s *DockerService) GetADIdentifiers() ([]string, error) {
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to inspect container %s", string(s.ID)[:12])
+	}
+	for net, settings := range cInspect.NetworkSettings.Networks {
+		ips[net] = settings.IPAddress
+	}
+	if _, found := ips["bridge"]; !found && cInspect.NetworkSettings.IPAddress != "" {
```
Comment (mfpierre) [inline pkg/collector/listeners/docker.go]: I think you can remove this chunk, `NetworkSettings.IPAddress` is deprecated

Comment (hkaj) [reply pkg/collector/listeners/docker.go]: dang you're right, and I gave you a card about this 🤦‍♂️

**Comment by hkaj** [comment]
Tests are failing because of the missing GetTags method. @mfpierre your PR will fix them. Merging

**Pull request merged**